client/my-sites/post-relative-time-status/index.jsx

```diff
@@ -21,118 +20,164 @@ class PostRelativeTime extends React.PureComponent {
 	static propTypes = {
 		showPublishedStatus: PropTypes.bool.isRequired,
 		post: PropTypes.object.isRequired,
-		includeNonDraftStatuses: PropTypes.bool,
 		link: PropTypes.string,
 		target: PropTypes.string,
 		gridiconSize: PropTypes.number,
 	};
 
 	static defaultProps = {
-		includeNonDraftStatuses: false,
 		link: null,
 		target: null,
 	};
 
+	/**
+	 * Get the date to be displayed depending on the status of the post
+	 */
 	getTimestamp() {
 		switch ( this.props.post.status ) {
 			case 'new':
 				return null;
+
 			case 'draft':
-			case 'future':
+			case 'trash':
 			case 'pending':
 				return this.props.post.modified;
+
 			default:
 				return this.props.post.date;
 		}
 	}
 
-	getDisplayedTimeFromPost( post ) {
+	getDisplayedTimeForLabel( noPrefix = false ) {
```

Comment:
Optional: negated variables are usually hard to follow in code. Using `prefix = true` here will likely make the code a bit more easier to understand.